### ISSUE TITLE
simultaneous: a character gets repeated while another key is pressed until simultaneous_threshold_milliseconds ends

### ISSUE THREAD

**Issue by diegobit** [tekezo/Karabiner-Elements#1290]
**Reason**: I'm trying to do SpaceFN and I noticed that, especially with higher `simultaneous_threshold_milliseconds` (with 50 it is not easy to press two keys simultaneously), sometimes keys pressed right after spacebar get repeated. 

**Explanation and steps to reproduce**: 
- use the complex_modifications below or set a high (like 1000) `basic.simultaneous_threshold_milliseconds`
- (as the example below) Press a letter, then immediately press spacebar, then release the letter but don't release spacebar, the letter gets repeated until `basic.simultaneous_threshold_milliseconds` ends.
```
letter l:       ----
spacebar:         --------------------------------------

EventViewer log: 
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x25       name:l               flags:                                   misc:
eventType:KeyDown         code:0x31       name:spacebar        flags:                                   misc:
eventType:KeyUp           code:0x25       name:l               flags:                                   misc:
eventType:KeyUp           code:0x31       name:spacebar        flags:                                   misc:

``` 
```
"complex_modifications": {
                "parameters": {
                    "basic.simultaneous_threshold_milliseconds": 1000,
                    "basic.to_delayed_action_delay_milliseconds": 500,
                    "basic.to_if_alone_timeout_milliseconds": 1000,
                    "basic.to_if_held_down_threshold_milliseconds": 500
                },
                "rules": [
                    {
                        "description": "SpaceFN layer",
                        "manipulators": [
                            {
                                "from": {
                                    "modifiers": {
                                        "optional": [
                                            "any"
                                        ]
                                    },
                                    "simultaneous": [
                                        {
                                            "key_code": "spacebar"
                                        },
                                        {
                                            "key_code": "l"
                                        }
                                    ],
                                    "simultaneous_options": {
                                        "key_down_order": "strict",
                                        "key_up_order": "strict_inverse",
                                        "to_after_key_up": [
                                            {
                                                "set_variable": {
                                                    "name": "SpaceFN",
                                                    "value": 0
                                                }
                                            }
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                        "set_variable": {
                                            "name": "SpaceFN",
                                            "value": 1
                                        }
                                    },
                                    {
                                        "key_code": "right_arrow"
                                    }
                                ],
                                "type": "basic"
                            },
                            {
                                "conditions": [
                                    {
                                        "name": "SpaceFN",
                                        "type": "variable_if",
                                        "value": 1
                                    }
                                ],
                                "from": {
                                    "key_code": "l",
                                    "modifiers": {
                                        "optional": [
                                            "any"
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                        "key_code": "right_arrow"
                                    }
                                ],
                                "type": "basic"
                            }
                        ]
                    }
                ]
            },
```  

Am I doing something wrong or it this a bug?

Karabiner version 11.6.5

**Comment by tekezo** [comment]
Thank you for feedback!
I confirmed this issue.
I'll fix this problem soon. Please wait.

**Comment by tekezo** [comment]
I fixed this issue at v11.6.6.
Please upgrade to the latest beta version.

**Comment by diegobit** [comment]
Perfect! I confirm I no longer experience the issue.
Thank you!
